### PR TITLE
Added gender_identity field and ontology. Fixes #1409.

### LINKED ISSUE #1409: New field: Gender Identity
**For which schema is a change/update being suggested?**

I would like a request for the `donor_organism.json` schema and include a new ontology module, `gender.json`

**What should the change/update be?**

I would like to add a new optional field, `gender_identity` to the donor_organism schema to allow for the correct assignment of gender when the information is available.

Along this update, I would like to create a new ontology module that will collect and validate against the terms below [gender](https://www.ebi.ac.uk/ols/ontologies/ncit/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FNCIT_C17357)

This update constitutes a minor change to the schema(s) it affects.

**What new field(s) need to be changed/added?**

* Field name: gender_identity
* Field description: The donor's personal sense of their own gender at the time of the experiment.
* Field type: Object (Ontology)
* Required: no
* Examples: (Not required due to loading an object)
* CV or enum: Ontology - `module/ontology/gender_identity.json`

**What new schema entity needs to be added?**

* Schema name: `module/ontology/gender_identity.json`
* Graph restrictions:
   * ontologies: ["obo:hcao", "obo:ncit"]
   * classes: ["NCIT:C17357"]
   * relation: ["rdfs:subClassOf"]
   * direct: false
   * include_self: false

**Why is the change requested?**

This update is essential to allow for gender identity, which may imply genetic differences on certain datasets.

### PR DESCRIPTION
**Pull request by ESapenaVentura**
### Release notes

For `type/biomaterial/donor_organism.json` schema:
- Added ontologised gender_identity field

For `module/gender_identity_ontology.json` schema:
- Created ontology module for terms under [Gender](https://www.ebi.ac.uk/ols/ontologies/ncit/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FNCIT_C17357&lang=en&viewMode=All&siblings=false)

### Reviews requested

- This is a major update for all dependencies of the new module (donor_organism)

### Rationale
This change was requested to support the metadata for the "Testis cell atlas" dataset, in which there are 2 testis samples from female-gendered individuals. It will also help wranglers in the future to be more inclusive for non-cis individuals.

### CODE REVIEW

**Comment by ESapenaVentura** [comment]
Please ignore the "schema linting" failed action, it is complaining that the ontology branch has not been yet imported to the HCAO.

The HCAO works by releases and this will be imported in our next release (~18th) https://github.com/HumanCellAtlas/ontology/issues/113

**Review by ncalvanese1 — APPROVED** [review]
Looks good to me!

**Review by NoopDog — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Review by amnonkhen — COMMENTED** [review]
a few comments

File: docs/jsonBrowser/type.md [hunk 1]
```diff
@@ -414,6 +414,7 @@ height_unit | The unit in which Height is expressed. | object | no | [See module
 weight | Weight of organism in Weight unit. | string | no |  | Weight |  | 60; 40-60
 weight_unit | The unit in which Weight is expressed. | object | no | [See module  mass_unit_ontology](module.md/#mass_unit_ontology) | Weight unit |  | 
 timecourse | Information relating to a timecourse associated with this biomaterial. | object | no | [See module  timecourse](module.md/#timecourse) | Timecourse |  | 
+gender_identity | The donor's personal sense of their own gender at the time of the experiment. | object | no | [See module  gender_identity_ontology](module.md/#gender_identity_ontology) | Gender identity |  |
```
[the remark this answers is not in the document]
Comment (ESapenaVentura) [reply docs/jsonBrowser/type.md:417]: Good catch, but this is generated automatically. I will open a ticket to fix the script called by the pre-commit hook that generates this

**Review by ruchim — APPROVED** [review]
looks good to me!

**Review by amnonkhen — APPROVED** [review]

**Comment by arschat** [comment]
This was previously blocked by an ontology update that is now completed.
Furthermore, Genetic Diversity network has requested the `genderidentity_selfreported` field, using the [NCIT:C17357](https://www.ebi.ac.uk/ols4/ontologies/ncit/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FNCIT_C17357) ontology branch.

This PR is updated with the GD network preferred ontology and merged current staging branch into this, so we are re-requesting reviews.

**Review by amnonkhen — APPROVED** [review]
LGTM

**Review by hannes-ucsc — APPROVED** [review]

**Review by NoopDog — APPROVED** [review]

**Pull request merged**